de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/filesystem/EclipseReferencePointImpl.java

```diff
@@ -0,0 +1,36 @@
+package de.fu_berlin.inf.dpp.filesystem;
+
+public class EclipseReferencePointImpl implements IReferencePoint {
+
+    final EclipsePathImpl path;
+    protected IReferencePoint referencePoint;
+
+    public EclipseReferencePointImpl(EclipsePathImpl path) {
```

Comment:
I do not get the implementation, you basically implement the reference point X times and the implementation are hashCode and equals, both methods completely auto-generated.

Reply:
The IReferencePoint implementations should be comparable by the IPath. So the equals method has to be overwritten.